lesson_71/HW_L71/controllers/product.controllers.js

```diff
@@ -49,4 +51,17 @@ export const getProductVideoByName = (req, res, next) => {
 
 export const getProductPreviewByName = (req, res, next) => {
   retrieveProductPreview(req, res, next, handleResponseGetMedia(res, next));
+};
+
+export const fetchProducts = async (req, res) => {
+  res.status(200).json(await getProducts());
+};
+
+export const fetchProductById = async (req, res, next) => {
+  const foundProductById = await findProductById(req.params);
+
+  if (!foundProductById) {
+    return next(new ErrorObjectNotFound("product not found"));
```

Comment:
you can use `throw new ErrorObjectNotFound` instead of next(), if you throw an error, it will be passed to error handling middleware even if next() wasn't invoked